### PR TITLE
driver: i2c: fix recovery mechanism in npcx series.

### PR DESCRIPTION
**Pull request by MulinChao**
Since the PU power rail of i2c bus might be gone at the initial state
after ec powered up, we might have no chance to get STOP condition. This
is because no i2c transactions occurred before its power rail is
restored. But it's crucial to reset the whole i2c module after i2c bus
is back to the idle state.

The original test suite for i2c recovery mechanism didn't consider this
case that initial i2c bus is low before ec powered on. Hence, this CL
fixed this symptom on Volteer by:

1. Force i2c modules must proceed 'reset' step no matter we received
   STOP condition or not.
2. Use Boolean for condition check to prevent misusage and meet MISRA.

### CODE REVIEW

File: drivers/i2c/i2c_npcx_controller.c [hunk 1]
```diff
@@ -414,6 +414,9 @@ static int i2c_ctrl_recovery(const struct device *dev)
 	struct i2c_ctrl_data *const data = DRV_DATA(dev);
 	int ret;
 
+	if (data->oper_state != NPCX_I2C_ERROR_RECOVERY)
```
Comment (pabigot) [inline drivers/i2c/i2c_npcx_controller.c:417]: braces around single statement body

**Review by pabigot — CHANGES_REQUESTED** [review]
Looks reasonable, with one style fix needed.

File: drivers/i2c/i2c_npcx_controller.c (hunk 1 above)
Comment (MulinChao) [reply drivers/i2c/i2c_npcx_controller.c]: Thanks. I will remind myself of this coding rule.

Comment (MulinChao) [reply drivers/i2c/i2c_npcx_controller.c]: Done.

**Review by nashif — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Pull request merged**